lib/engine/game/g_18_ny/game.rb

```diff
@@ -257,37 +316,50 @@ def event_close_companies!
 
         def event_float_30!
           @log << "-- Event: #{EVENTS_TEXT['float_30'][1]} --"
-          non_floated_companies { |c| c.float_percent = 30 }
+          @float_percent = 30
+          non_floated_corporations { |c| c.float_percent = @float_percent }
         end
 
         def event_float_40!
           @log << "-- Event: #{EVENTS_TEXT['float_40'][1]} --"
-          non_floated_companies { |c| c.float_percent = 40 }
+          @float_percent = 40
+          non_floated_corporations { |c| c.float_percent = @float_percent }
         end
 
         def event_float_50!
           @log << "-- Event: #{EVENTS_TEXT['float_50'][1]} --"
-          non_floated_companies { |c| c.float_percent = 50 }
+          @float_percent = 50
+          non_floated_corporations { |c| c.float_percent = @float_percent }
         end
 
         def event_float_60!
           @log << "-- Event: #{EVENTS_TEXT['float_60'][1]} --"
-          non_floated_companies do |c|
-            c.float_percent = 60
+          @float_percent = 50
+          non_floated_corporations do |c|
+            c.float_percent = @float_percent
             c.capitalization = :full
             c.spend(c.cash, @bank) if c.cash.positive?
           end
         end
 
         def event_nyc_formation!
+          return if @nyc_formation_state
+
           @log << "-- Event: #{EVENTS_TEXT['nyc_formation'][1]} --"
+          @nyc_formation_state = :round_one
+
+          @log << 'No further minor corporations may be started'
+          @corporations.select { |c| c.type == :minor && !c.floated? && !c.closed? }.each do |c|
```

Comment:
remove the select and just do an next if in the each

Reply:
Fixed